xclim/sdba/adjustment.py

```diff
@@ -406,6 +459,168 @@ def _adjust(self, sim, interp="nearest", extrapolation="constant"):
+        the final extreme value :math:`V_i` is:
+
+        .. math::
+
+            V_i = C_i * \tau + S_i * (1 - \tau)
+
+        Where :math:`\tau` is a function of sim's extreme values :math:`F` and of arguments
+        ``frac`` (:math:`f`) and ``power`` (:math:`p`):
+
+        .. math::
+
+            \tau = \left(\frac{1}{f}\frac{S - min(S)}{max(S) - min(S)}\right)^p
+
+        References
+        ----------
+        .. [ClimateTools] https://juliaclimate.github.io/ClimateTools.jl/stable/
+        .. [RRJF2021] Roy, P., Rondeau-Genesse, G., Jalbert, J., Fournier, É. 2021. Climate Scenarios of Extreme Precipitation Using a Combination of Parametric and Non-Parametric Bias Correction Methods. Submitted to Climate Services, April 2021.
+        """
+        super().__init__(q_thresh=q_thresh, cluster_thresh=cluster_thresh, dist=dist)
+
+    def _train(self, ref, hist):
+        cluster_thresh = convert_units_to(self.cluster_thresh, ref)
+        hist = convert_units_to(hist, ref)
+
+        # Extreme value threshold computed relative to "large values".
+        # We use the mean between ref and hist here.
+        thresh = (
+            ref.where(ref >= cluster_thresh).quantile(self.q_thresh, dim="time")
+            + hist.where(hist >= cluster_thresh).quantile(self.q_thresh, dim="time")
+        ) / 2
+
+        # All large value clusters
+        ref_clusters = get_clusters(ref, thresh, cluster_thresh)
+        # Parameters of a genpareto (or other) distribution, we force te location at thresh.
+        fit_params = stats.fit(
+            ref_clusters.maximum - thresh, self.dist, dim="cluster", floc=0
+        )
+        # Param "loc" was fitted with 0, put thresh back
+        fit_params[..., 1] = fit_params[..., 1] + thresh
+
+        ds = xr.Dataset(
+            dict(fit_params=fit_params, thresh=thresh, nclusters=ref_clusters.nclusters)
+        )
+        ds.fit_params.attrs.update(
+            long_name=f"{self.dist} distribution parameters of ref",
+        )
+        ds.thresh.attrs.update(
+            long_name=f"{self.q_thresh * 100}th percentile extreme value threshold",
+            description=f"Mean of the {self.q_thresh * 100}th percentile of large values (x > {self.cluster_thresh}) of ref and hist.",
+        )
+        self.set_dataset(ds)
+
+    def _adjust(
+        self,
+        scen,
+        sim,
+        frac: float = 0.25,
+        power: float = 1.0,
+    ):
+        def _adjust_extremes_1d(scen, sim, ref_params, thresh, *, dist, cluster_thresh):
+            # Clusters of large values of sim
+            _, _, sim_posmax, sim_maxs = get_clusters_1d(sim, thresh, cluster_thresh)
+
+            new_scen = scen.copy()
+            if sim_posmax.size == 0:
+                # Happens if everything is under `cluster_thresh`
+                return new_scen
+
+            # Fit the dist, force location at thresh
+            sim_fit = stats._fitfunc_1d(
```

Comment:
I guess this assumes a detrended sim series ? Worth mentioning in the docstring ?